src/arviz_plots/plots/violin_plot.py

```diff
@@ -0,0 +1,331 @@
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    if backend is None:
+        if plot_collection is None:
+            backend = rcParams["plot.backend"]
+        else:
+            backend = plot_collection.backend
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+
+    if plot_collection is None:
+        if backend is None:
+            backend = rcParams["plot.backend"]
+        pc_kwargs["figure_kwargs"] = pc_kwargs.get("figure_kwargs", {}).copy()
+        pc_kwargs.setdefault(
+            "cols",
+            ["__variable__"]
+            + [dim for dim in distribution.dims if dim not in {"model"}.union(sample_dims)],
+        )
+        if "model" in distribution:
+            pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+            pc_kwargs["aes"].setdefault("color", ["model"])
+            pc_kwargs["aes"].setdefault("x", ["model"])
+
+        pc_kwargs = set_wrap_layout(pc_kwargs, plot_bknd, distribution)
+        plot_collection = PlotCollection.wrap(
+            distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+
+    if aes_by_visuals is None:
+        aes_by_visuals = {}
+    else:
+        aes_by_visuals = aes_by_visuals.copy()
+    aes_by_visuals.setdefault("left_density", plot_collection.aes_set.difference("x"))
+    aes_by_visuals.setdefault("right_density", plot_collection.aes_set.difference("x"))
+    if "model" in distribution:
+        aes_by_visuals.setdefault("credible_interval", ["color", "x"])
+        aes_by_visuals.setdefault("point_estimate", ["color", "x"])
+    if "point_estimate" in aes_by_visuals and "point_estimate_text" not in aes_by_visuals:
+        aes_by_visuals["point_estimate_text"] = aes_by_visuals["point_estimate"]
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # density
+    left_density_kwargs = copy(visuals.get("left_density", {}))
+
+    if left_density_kwargs is not False:
+        left_density_dims, left_density_aes, left_density_ignore = filter_aes(
+            plot_collection, aes_by_visuals, "left_density", sample_dims
+        )
+        default_color = plot_bknd.get_default_aes("color", 1, {})[0]
+        if "color" not in left_density_aes:
+            left_density_kwargs.setdefault("color", default_color)
+        with warnings.catch_warnings():
+            if "model" in distribution:
+                warnings.filterwarnings("ignore", message="Your data appears to have a single")
+            left_density = distribution.azstats.kde(
+                dim=left_density_dims, **stats.get("left_density", {})
+            )
+
+        # flip x and y coordinates
+        density_flipped = left_density.rename({"plot_axis": "flipped_plot_axis"})
+        density_flipped = density_flipped.assign_coords(flipped_plot_axis=["y", "x"])
+        density_flipped = density_flipped.rename({"flipped_plot_axis": "plot_axis"})
```

Comment:
```suggestion
        # flip x and y coordinates
        density_flipped = left_density.sel(plot_axis=["x", "y"]).assign_coords(plot_axis=["y", "x"])
```

This should achieve the same, and adding the `.sel` will make it work even if for some reason the initial dataarray already has y, x as coord values